src/python/grpcio_tests/tests_aio/unit/compression_test.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2020 The gRPC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests behavior around the compression mechanism."""
+
+import asyncio
+import logging
+import platform
+import random
+import unittest
+
+import grpc
+from grpc.experimental import aio
+
+from tests_aio.unit._test_base import AioTestBase
+from tests_aio.unit import _common
+
+_GZIP_CHANNEL_ARGUMENT = ('grpc.default_compression_algorithm', 2)
+_GZIP_DISABLED_CHANNEL_ARGUMENT = ('grpc.compression_enabled_algorithms_bitset',
+                                   3)
+_DEFLATE_DISABLED_CHANNEL_ARGUMENT = (
+    'grpc.compression_enabled_algorithms_bitset', 5)
+
+_TEST_UNARY_UNARY = '/test/TestUnaryUnary'
+_TEST_SET_COMPRESSION = '/test/TestSetCompression'
+_TEST_DISABLE_COMPRESSION_UNARY = '/test/TestDisableCompressionUnary'
+_TEST_DISABLE_COMPRESSION_STREAM = '/test/TestDisableCompressionStream'
+
+_REQUEST = b'\x01' * 100
+_RESPONSE = b'\x02' * 100
+
+
+async def _test_unary_unary(unused_request, unused_context):
+    return _RESPONSE
+
+
+async def _test_set_compression(unused_request_iterator, context):
+    assert _REQUEST == await context.read()
+    context.set_compression(grpc.Compression.Deflate)
+    await context.write(_RESPONSE)
+    try:
+        context.set_compression(grpc.Compression.Deflate)
+    except RuntimeError:
+        pass
```

Comment:
Will it be worth it to add a comment here saying what we have tested?

Reply:
Comment added:
```
        # NOTE(lidiz) Testing if the servicer context raises exception when
        # the set_compression method is called after initial_metadata sent.
        # After the initial_metadata sent, the server-side has no control over
        # which compression algorithm it should use.
```